src/metabase/models/dashboard.clj

```diff
@@ -226,3 +228,32 @@
         (dashboard-card/update-dashboard-card! (update dashboard-card :series #(filter identity (map :id %))))))
     (let [new-param-field-ids (dashboard-id->param-field-ids dashboard-or-id)]
       (update-field-values-for-on-demand-dbs! dashboard-or-id old-param-field-ids new-param-field-ids))))
+
+(defn- save-card!
+  [card]
+  (db/insert! 'Card
+    (-> card
+        (dissoc :id)
+        (update :result_metadata #(or % (-> card
+                                            :dataset_query
+                                            card.api/result-metadata-for-query))))))
+
+(defn save-transient-dashboard!
```

Comment:
Seems like we should be wrapping this in a transaction. By default we autocommit everything, so if there was some failure in `save-card!`, we could end up with half a dashboard, right?